src/software/ai/hl/stp/play/penalty_kick/penalty_kick_play_fsm.cpp

```diff
@@ -0,0 +1,74 @@
+#include "software/ai/hl/stp/play/penalty_kick/penalty_kick_play_fsm.h"
+
+PenaltyKickPlayFSM::PenaltyKickPlayFSM(TbotsProto::AiConfig ai_config)
+    : ai_config(ai_config),
+      penalty_kick_tactic(std::make_shared<PenaltyKickTactic>(ai_config)),
+      penalty_setup_tactics(std::vector<std::shared_ptr<PenaltySetupTactic>>()),
+      away_from_kick_tactics(std::vector<std::shared_ptr<StopTactic>>())
+{
+}
+
+void PenaltyKickPlayFSM::performKick(const Update &event)
+{
+    PriorityTacticVector tactics_to_run = {{}};
+
+    tactics_to_run[0].insert(tactics_to_run[0].end(), away_from_kick_tactics.begin(),
+                             away_from_kick_tactics.end());
+    tactics_to_run[0].emplace_back(penalty_kick_tactic);
+    event.common.set_tactics(tactics_to_run);
+}
+
+void PenaltyKickPlayFSM::setupPosition(const Update &event)
+{
+    PriorityTacticVector tactics_to_run = {{}};
+
+    Vector behind_ball_direction = (event.common.world.ball().position() -
+                                    event.common.world.field().enemyGoalpostPos())
+                                       .normalize();
+    Angle shoot_angle = (event.common.world.field().enemyGoalpostPos() -
+                         event.common.world.ball().position())
+                            .orientation();
+
+    Point behind_ball = event.common.world.ball().position() +
+                        behind_ball_direction.normalize(DIST_TO_FRONT_OF_ROBOT_METERS +
+                                                        BALL_MAX_RADIUS_METERS + 0.1);
+    double ball_position_x = event.common.world.field().friendlyPenaltyMark().x();
+
+    // Adjust number of tactics based on the number of robots available
+    unsigned int num_tactics = event.common.num_tactics;
+    if (num_tactics != away_from_kick_tactics.size())
```

Comment:
doesn't look like we use away_from_kick_tactics (we never add anything to it)

Reply:
You're right, I actually don't need it